### PR TITLE
Migrate xpallets-assets-registrar to pallet v2

### PR DESCRIPTION
**Pull request by VegeBun-csj**

### CODE REVIEW

File: xpallets/assets-registrar/src/lib.rs [hunk 1]
```diff
@@ -19,118 +19,59 @@ pub mod weights;
...
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
 
-decl_storage! {
-    trait Store for Module<T: Config> as XAssetsRegistrar {
-        /// Asset id list for each Chain.
-        pub AssetIdsOf get(fn asset_ids_of): map hasher(twox_64_concat) Chain => Vec<AssetId>;
+    #[pallet::config]
+    /// The module's config trait.
```
Comment (0xrjman) [inline xpallets/assets-registrar/src/lib.rs]:
```suggestion
    /// The pallet's config trait.
```

File: xpallets/assets-registrar/src/lib.rs [hunk 2]
```diff
@@ -219,13 +162,104 @@ decl_module! {
...
             }
-            AssetInfoOf::insert(id, info);
+            AssetInfoOf::<T>::insert(id, info);
             Ok(())
         }
     }
+
+    #[pallet::event]
+    #[pallet::generate_deposit(pub(super) fn deposit_event)]
+    /// Event for the XAssetRegistrar Pallet
```
Comment (0xrjman) [inline xpallets/assets-registrar/src/lib.rs]:
```suggestion
    /// Event for the XAssetRegistrar Pallet
    #[pallet::event]
    #[pallet::generate_deposit(pub(super) fn deposit_event)]
```

**Review by 0xrjman — COMMENTED** [review]
I suggest rearranging the order of pallet components as implied in the upgrade guide:
`#[pallet::config]`
`#[pallet::pallet]`
`#[pallet::call]`
...

File: xpallets/assets-registrar/src/lib.rs [hunk 3]
```diff
@@ -300,17 +334,30 @@ impl<T: Config> Module<T> {
...
+    /// Direct implementation of `GenesisBuild::assimilate_storage`.
+    ///
+    /// Kept in order not to break dependency.
+    pub fn assimilate_storage<T: Config>(
+        &self,
+        storage: &mut sp_runtime::Storage,
+    ) -> Result<(), String> {
+        <Self as GenesisBuild<T>>::assimilate_storage(self, storage)
+    }
+}
```
Comment (liuchengxu) [inline xpallets/assets-registrar/src/lib.rs]: This is not neccessary, the compile issue should be resolved in another way.

```suggestion
```

Comment (alannotnerd) [reply xpallets/assets-registrar/src/lib.rs]: impl `GenesisBuild` in std

**Pull request merged**